js/zb.js

```diff
@@ -969,21 +969,88 @@ module.exports = class zb extends Exchange {
+        const method = this.getSupportedMapping (market['type'], {
+            'spot': 'spotV1PrivateGetOrder',
+            'swap': 'contractV2PrivatePostTradeOrder',
+        });
         const request = {
-            'price': this.priceToPrecision (symbol, price),
             'amount': this.amountToPrecision (symbol, amount),
-            'tradeType': (side === 'buy') ? '1' : '0',
-            'currency': this.marketId (symbol),
-        };
-        const response = await this.spotV1PrivateGetOrder (this.extend (request, params));
-        return {
-            'info': response,
-            'id': response['id'],
         };
+        if (price) {
+            request['price'] = this.priceToPrecision (symbol, price);
+        }
+        if (spot) {
+            request['tradeType'] = (side === 'buy') ? '1' : '0';
+            request['currency'] = market['id'];
+        } else if (swap) {
+            const reduceOnly = this.safeValue (params, 'reduceOnly');
+            if (side === 'sell' && reduceOnly) {
+                request['side'] = 3; // close long
+            } else if (side === 'buy' && reduceOnly) {
+                request['side'] = 4; // close short
+            } else if (side === 'buy') {
+                request['side'] = 1; // open long
+            } else if (side === 'sell') {
+                request['side'] = 2; // open short
+            }
+            if (type === 'limit') {
+                request['action'] = 1;
+            } else if (timeInForce === 'IOC') {
+                request['action'] = 3;
+            } else if (timeInForce === 'PO') {
+                request['action'] = 4;
+            } else if (timeInForce === 'FOK') {
+                request['action'] = 5;
+            } else {
+                request['action'] = type;
+            }
+            request['symbol'] = market['id'];
+            request['clientOrderId'] = params['clientOrderId']; // OPTIONAL '^[a-zA-Z0-9-_]{1,36}$', // The user-defined order number
+            request['extend'] = params['extend']; // OPTIONAL {"orderAlgos":[{"bizType":1,"priceType":1,"triggerPrice":"70000"},{"bizType":2,"priceType":1,"triggerPrice":"40000"}]}
+        }
+        let response = await this[method] (this.extend (request, params));
+        //
+        // Spot
+        //
+        //     {
+        //         "code": 1000,
+        //         "message": "操作成功",
+        //         "id": "202202224851151555"
+        //     }
+        //
+        // Swap
+        //
+        //     {
+        //         "code": 10000,
+        //         "desc": "操作成功",
+        //         "data": {
+        //             "orderId": "6901786759944937472",
+        //             "orderCode": null
+        //         }
+        //     }
+        //
+        if (swap) {
+            response = this.safeValue (response, 'data');
+            response['timeInForce'] = timeInForce;
+        } else {
+            response['type'] = request['tradeType'];
+        }
```

Comment:
I don't see any need to restrict `timeInForce` to `swap` or `type` to `spot`, I think you may as well take these out of the if statement, so this would be

```
if (swap) {
      response = this.safeValue (response, 'data');   
}
response['timeInForce'] = timeInForce;
response['type'] = request['tradeType'];
```